libdevcore/CommonData.h

```diff
@@ -42,22 +42,24 @@ enum class WhenError
 	Throw = 1,
 };
 
-enum class HexPrefix
-{
-	DontAdd = 0,
-	Add = 1,
-};
+namespace {
+	const char* hexdigits = "0123456789abcdef";
+}
+
 /// Convert a series of bytes to the corresponding string of hex duplets.
-/// @param _w specifies the width of the first of the elements. Defaults to two - enough to represent a byte.
 /// @example toHex("A\x69") == "4169"
 template <class T>
-std::string toHex(T const& _data, int _w = 2, HexPrefix _prefix = HexPrefix::DontAdd)
+std::string toHex(T const& _data)
 {
-	std::ostringstream ret;
-	unsigned ii = 0;
-	for (auto i: _data)
-		ret << std::hex << std::setfill('0') << std::setw(ii++ ? 2 : _w) << (int)(typename std::make_unsigned<decltype(i)>::type)i;
-	return (_prefix == HexPrefix::Add) ? "0x" + ret.str() : ret.str();
+	size_t len = _data.size() * sizeof(typename T::value_type);
+	const byte* data = reinterpret_cast<const byte*>(_data.data());
```

Comment:
Can you tell me how to do it?

The problem is that `toHex` is used with many different types: vectors, vector_refs, arrays, strings etc. Ideally we'd have a way of saying that `data()` should return a byte-sized type. We cannot say `data` returns `byte` because it breaks strings. I arrived at this because it felt like the easiest solution and also works with stuff like `std::vector<uint32_t>`.